### DISCUSSION TITLE
How to add custome core/vanila javascript in nuxt 3 pages

### DISCUSSION THREAD

**Discussion by hirenramoliyaiwb** [nuxt/nuxt#16481] [Questions]
### Environment

success Install finished in 2.199s

Nuxi 3.0.0-rc.12
RootDir: /home/projects/mkliynxxq.github
Nuxt project info: 

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.12`
- Nitro Version:    `0.6.0`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

👉 Report an issue: https://github.com/nuxt/framework/issues/new

👉 Suggest an improvement: https://github.com/nuxt/framework/discussions/new

👉 Read documentation: https://v3.nuxtjs.org

### Reproduction

[https://stackblitz.com/github/nuxt/starter/tree/v3-stackblitz?file=app.vue](https://stackblitz.com/github/nuxt/starter/tree/v3-stackblitz?file=app.vue)

### Describe the bug

I want add custom javascript code for animation in any pages so I am not able to find any way @danielroe

**Comment by TheAlexLichter** [comment]
Please don't tag maintainers (unless you were in contact before and they asked you to).

Also, your reproduction does not show anything besides the default app. Please show what you tried.

You can e.g. add custom JavaScript via `useHead` (3rd party scripts and inline scripts) or put it in `app.vue` via `onMounted`. But without the code or some example it is hard to guess what you are trying to achieve.

**Reply by hirenramoliyaiwb** [reply]
I want to add this javascript code in index page in nuxt 3

var TxtType = function(el, toRotate, period) {
        this.toRotate = toRotate;
        this.el = el;
        this.loopNum = 0;
        this.period = parseInt(period, 10) || 2000;
        this.txt = '';
        this.tick();
        this.isDeleting = false;
    };

    TxtType.prototype.tick = function() {
        var i = this.loopNum % this.toRotate.length;
        var fullTxt = this.toRotate[i];

        if (this.isDeleting) {
        this.txt = fullTxt.substring(0, this.txt.length - 1);
        } else {
        this.txt = fullTxt.substring(0, this.txt.length + 1);
        }

        this.el.innerHTML = '<span class="wrap">'+this.txt+'</span>';

        var that = this;
        var delta = 200 - Math.random() * 100;

        if (this.isDeleting) { delta /= 2; }

        if (!this.isDeleting && this.txt === fullTxt) {
        delta = this.period;
        this.isDeleting = true;
        } else if (this.isDeleting && this.txt === '') {
        this.isDeleting = false;
        this.loopNum++;
        delta = 500;
        }

        setTimeout(function() {
        that.tick();
        }, delta);
    };

    window.onload = function() {
        var elements = document.getElementsByClassName('typewrite');
        for (var i=0; i<elements.length; i++) {
            var toRotate = elements[i].getAttribute('data-type');
            var period = elements[i].getAttribute('data-period');
            if (toRotate) {
              new TxtType(elements[i], JSON.parse(toRotate), period);
            }
        }
        // INJECT CSS
        var css = document.createElement("style");
        css.type = "text/css";
        css.innerHTML = ".typewrite > .wrap { border-right: 0.08em solid #fff}";
        document.body.appendChild(css);
    };

**Reply by TheAlexLichter** [reply]
You can add that both ways, either via [`useHead`](https://github.com/vueuse/head) or after `onMounted` (e.g. in an extra function).

Anyhow I'd recommend to do that more in a "vue-way" by writing components instead using `innerHTML` and other dom operations. It should not be too difficult to convert that instead of copy-pasting the [SO code](https://stackoverflow.com/questions/42144095/pausing-js-typewriter)
